### PR TITLE
fix(QSeparator): do not force width or height to avoid overflowing the parent #8307

### PR DESCRIPTION
**Pull request by pdanpdan**
Also exclude a few styles from RTL

#8307

### CODE REVIEW

File: ui/src/css/core/flex.sass [hunk 1]
```diff
@@ -135,9 +135,9 @@
       > *
         margin-left: $size
     &-y-#{$name}
-      margin-top: -$size
+      margin-top: -$size #{"/* rtl:ignore */"}
```
Comment (rstoenescu) [inline ui/src/css/core/flex.sass:138]: any non horizontal related CSS does not gets translated for RTL purposes, so this addition should in theory not matter

Comment (pdanpdan) [reply ui/src/css/core/flex.sass:138]: I know it should not have been processed, but unfortunately it is.

Comment (rstoenescu) [reply ui/src/css/core/flex.sass:138]: Just checked. It doesn't process it. Those lines are not getting doubled.

Comment (pdanpdan) [reply ui/src/css/core/flex.sass:138]: This is an extract from `quasar.rtl.css`:

```css
[dir] .q-gutter-y-none, [dir] .q-gutter-none {
  margin-top: 0;
}
[dir] .q-gutter-y-none > *, [dir] .q-gutter-none > * {
  margin-top: 0;
}
```

Comment (rstoenescu) [reply ui/src/css/core/flex.sass:138]: The "dir" attribute is assigned to the `<html>` tag by Quasar. So it's fine.

Comment (pdanpdan) [reply ui/src/css/core/flex.sass:138]: I know, but a rule with dir gets higher priority than one without. So a child of .q-gutter-y-md cannot have it's vertical padding changed by a .q-my-lg

**Pull request merged**